### ISSUE TITLE
Add Storybook story variants for `Button` states (and VRT scenarios)

### ISSUE THREAD

**Issue by nfmohit** [google/site-kit-wp#8054]
## Feature Description

We've got Storybook stories for the `Button` component, however, similar to [the Figma designs](https://www.figma.com/file/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?type=design&node-id=409-2253&mode=design&t=bS7348LNGZTkPUen-0), we do not have the different button states (e.g. hover, focus, pressed, disabled, etc.) covered in Storybook or VRT scenarios.

We should update the Storybook stories for the `Button` component so that these states are also covered.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The Button component stories should have the additional states added for each button variant:
   * focussed (this is considered the same as [pressed in the Material Design 3 guidelines](https://m3.material.io/foundations/interaction/states/overview))
   * disabled
* These new states should be included within the All Buttons VRT story.
* The Default Button Hover story shows a genuinely hovered button, matching how the button looks on hover in the browser.

## Implementation Brief

* [x] In `storybook/package.json`:
   * Add `storybook-addon-pseudo-states` to `devDependencies`, at the version that supports Storybook 8.6.

* [x] In `storybook/main.js`:
   * Register the addon in the `addons` array with `getModuleAbsolutePath( 'storybook-addon-pseudo-states' )`.

* [x] In `assets/js/googlesitekit/components-gm2/Button/index.stories.js`:
   * Drop the `googlesitekit-button--hover` class from `HoverButton`. It is not defined in any stylesheet, so it renders a plain default button.
   * Give `HoverButton` an `args.className` of `googlesitekit-vrt-button-hover` and a `parameters.pseudo` of `{ hover: true }`, following the `googlesitekit-vrt-` naming already used for VRT-only hooks in `storybook/preview-head-vrt.html`.
   * Add a focus story for each variant, named `Default Button Focus`, `Default Button Link Focus`, `Danger Button Focus`, `Tertiary Button Focus`, `Icon Button Focus`, `Callout Button Focus`, `Callout Warning Button Focus` and `Callout Error Button Focus`, each reusing its variant's existing args.
   * Give every focus story an `args.className` of `googlesitekit-vrt-button-focus mdc-ripple-upgraded--background-focused` and a `parameters.pseudo` of `{ focus: true }`.
   * `mdc-ripple-upgraded--background-focused` is required as well as the pseudo class: `Button` attaches `MDCRipple`, and MDC paints the focus state layer for an upgraded ripple off that class rather than off `:focus`, so the raised, danger and icon variants would otherwise render identically to their default state.
   * Add `Disabled Danger Button`, `Disabled Tertiary Button` and `Disabled Icon Button` stories, matching the naming of the existing `Disabled Callout Button` stories.
   * Add `HoverButton` and every new story to the `buttonStories` array in `VRTStory`, grouping each variant's default, hover, focus and disabled entries together.
   * Set `VRTStory.parameters` to `{ pseudo: { hover: [ '.googlesitekit-vrt-button-hover' ], focus: [ '.googlesitekit-vrt-button-focus' ] } }` so all marked buttons render their state in the single screenshot.
   * Leave `VRTStory.scenario` unchanged: the icon button's tooltip is opened by a real pointer hover, not by CSS.

### Test Coverage

* Storybook stories in `assets/js/googlesitekit/components-gm2/Button/index.stories.js`:
   * A focus story for each of the eight variants, rendering the variant's focus background and state layer.
   * A disabled story for the danger, tertiary and icon variants, alongside the existing disabled default and callout stories.
   * `Default Button Hover` rendering the hover state layer rather than a default button.
   * `All Buttons VRT` showing every variant's default, focus and disabled state together, plus the hovered default button and the icon button's open tooltip.
* The existing reference image for the `Components/Button/All Buttons VRT` scenario changes and needs re-approving.

## QA Brief

- Check the `Button` stories in [storybook](https://storage.googleapis.com/site-kit-github-artifacts/storybook/develop/index.html?path=/story/components-button--vrt-story).
- Ensure that it has additonal states for `hover` and `disabled`.

## Changelog entry

* N/A.

**Comment by benbowler** [comment]
Note that the hover state is mentioned in the issue description but this state already exists in the button stories.

**Comment by techanvil** [comment]
Hey @benbowler, thanks for drafting these AC. A couple of points:

- Although, as you've noted in your comment above, we do have a [Default Hover Button](https://storage.googleapis.com/site-kit-storybook/develop/index.html?path=/story/components-button--hover-button) story, it doesn't actually render a hovered button. It applies the `googlesitekit-button--hover` class, but I think this is an old style from the Material Design 2 days that has been removed. We do need to update this story as part of this issue.
- Further to the above, you've referenced the Material Design 2 documentation in the AC, but we should be referring to the [Material Design 3](https://m3.material.io/foundations/interaction/states/overview) guidelines. Although we haven't yet migrated to off-the-shelf MD3 components, we have applied MD3 style overrides to our MD2 components, and all our contemporary UX design uses MD3 as a reference.

**Comment by benbowler** [comment]
Thanks @techanvil, both done.

On the hover story, one correction: `googlesitekit-button--hover` has never had a definition in any stylesheet here, so it isn't a removed MD2 style. It's a Backstop `hoverSelector` hook, same as `googlesitekit-cta-link--hover` in `Link.stories.js`. It's doubly dead in this file because the VRT scenario hovers `.googlesitekit-button--icon` and the story was never added to `buttonStories`.

The IB now adds `storybook-addon-pseudo-states`, which rewrites the real `:hover` and `:focus` rules into class selectors, so the stories show what the browser shows and no test-only CSS ships to production.

One gotcha for whoever picks this up: `Button` attaches `MDCRipple`, and MDC paints the focus state layer off `.mdc-ripple-upgraded--background-focused` rather than `:focus`, so without that class the raised, danger and icon focus stories render identically to their default state.

AC updated to reference the MD3 states guidelines.

Also opened #13508 for the same problem in the Link stories, where seven hover variants render unhovered.

**Comment by techanvil** [comment]
Hi @benbowler, thanks for the clarification and the update. Both the AC & IB LGTM ✅

**Comment by mohitwp** [comment]
### QA Update ⚠️ 

Verified the Button component Storybook stories and All Buttons VRT story against the acceptance criteria.

@ankitrox  Please find couple of observations listed below -


**Observed:**
⚠️ **Focused state:**
I could not find any focused (pressed) state variants in the individual button stories or within the All Buttons VRT story.

As per the acceptance criteria:
"focussed (this is considered the same as pressed in the Material Design 3 guidelines)"
Could you please confirm if the focused/pressed states are expected to be added for each button variant as part of this ticket?

⚠️ **Default Button Hover story:**
The story loads with the default button state. The hover styling is applied only after manually hovering over the button with the cursor.

Expected:
The "Default Button Hover" story should display the button already in the hover state to represent a genuinely hovered button, matching the browser hover appearance.

Could you please confirm if the hover state is expected to be forced on initial story load?


<details>

<img width="893" height="631" alt="Image" src="https://github.com/user-attachments/assets/944b5639-9ff2-4af6-b209-e1016c95443e" />

<img width="1049" height="783" alt="Image" src="https://github.com/user-attachments/assets/5ce1f62d-bee9-4183-8dcd-0216917848f6" />

</details>


**Verified:**
✅ All Buttons VRT story includes the available button variants:
- Default Button
- Default Button Link
- Danger Button
- Callout Button
- Callout Warning Button
- Callout Error Button
- Tertiary Button
- Icon Button with Tooltip

✅ Disabled states are added and included:
- Disabled Button
- Disabled Callout Button
- Disabled Callout Error Button
- Disabled Callout Warning Button

**Comment by ankitrox** [comment]
@mohitwp Thank you for looking into this issue.

⚠️ **Focused State**

This is correct as the material UI considers focus as the pressed one, we do not need the separate pressed variant in the story.

⚠️ Hover story for default button

The hover [variant](https://storage.googleapis.com/site-kit-github-artifacts/storybook/develop/index.html?path=/story/components-button--hover-button&globals=pseudo.hover:!true) for default button seems correct to me. Please refer to attached video where it's styling is different from default state and also the state is shows as hovered in the story in the top bar.

<details>

https://github.com/user-attachments/assets/fa50d09e-51e3-403d-b1c1-c000bc6ff0aa

</details>

Please let me know in case of any further questions.

Thank you.

**Comment by mohitwp** [comment]
### QA Update ✅

Verified the Button component Storybook stories and the All Buttons VRT story against the acceptance criteria.

Also retested the observations raised earlier after the clarification:

✅ Focused state is covered as expected; a separate pressed variant is not required.
✅ Default Button Hover story correctly represents the hover state.
✅ Available button variants and their corresponding state stories are displayed as expected in Storybook.
✅ All Buttons VRT story includes the expected button variants/states.
✅ Button styling and state behaviour are working as expected.

**Verified:**
✅ All Buttons VRT story includes the available button variants:
- Default Button
- Default Button Link
- Danger Button
- Callout Button
- Callout Warning Button
- Callout Error Button
- Tertiary Button
- Icon Button with Tooltip

✅ Disabled states are added and included:
- Disabled Button
- Disabled Callout Button
- Disabled Callout Error Button
- Disabled Callout Warning Button
